apps/web/src/pages/integrations/components/multi-provider/UpdateProviderSidebar.tsx

```diff
@@ -195,20 +197,27 @@ export function UpdateProviderSidebar({
 
     const hasConditionsAndIsPrimary = hasUpdatedConditions && primary && dirtyFields.conditions;
 
-    if (
-      (hasNoConditions && isChangedToActive) ||
-      isChangedToInactiveAndIsPrimary ||
-      isPrimaryAndHasConditionsApplied ||
-      hasConditionsAndIsPrimary
-    ) {
-      openSelectPrimaryIntegrationModal({
+    const selectPrimaryIntegration = () => {
+      return openSelectPrimaryIntegrationModal({
         environmentId: selectedProvider?.environmentId,
         channelType: selectedProvider?.channel,
         exclude: !isActive || hasConditionsAndIsPrimary ? (el) => el._id === selectedProvider.integrationId : undefined,
         onClose: () => {
           updateIntegration(data);
         },
       });
+    };
```

Comment:
I decided to move calling `openSelectPrimaryIntegrationModal` to function, because it will be called multiple times with same arguments.